### PR TITLE
Feature discovery for new Noviflow switches

### PR DESCRIPTION
**Pull request by IlyaMoiseev**

### CODE REVIEW

File: services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/AbstractFeature.java [hunk 1]
```diff
@@ -17,13 +17,16 @@
...
+import com.google.common.collect.Sets;
 import net.floodlightcontroller.core.IOFSwitch;
 
 import java.util.Optional;
+import java.util.Set;
 
 public abstract class AbstractFeature {
     protected static final String MANUFACTURER_NICIRA = "Nicira, Inc.";
-    protected static final String MANUFACTURER_NOVIFLOW = "NoviFlow Inc";
+    protected static final Set<String> MANUFACTURER_DESCRIPTIONS_NOVIFLOW =
```
Comment (surabujin) [inline services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/AbstractFeature.java]: `_DESCRIPTIONS_` is extra part here.

File: services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/BfdFeature.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
         if (description == null) {
             return empty;
         }
-        if (!MANUFACTURER_NOVIFLOW.equals(description.getManufacturerDescription())) {
+        if (!MANUFACTURER_DESCRIPTIONS_NOVIFLOW.contains(description.getManufacturerDescription())) {
```
Comment (surabujin) [inline services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/BfdFeature.java]: And where is NW440 regexp?

File: services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/AbstractFeature.java (hunk 1 above)
Comment (IlyaMoiseev) [reply services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/AbstractFeature.java]: Keep variable name as it was

File: services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/BfdFeature.java (hunk 2 above)
Comment (IlyaMoiseev) [reply services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/BfdFeature.java]: Changed to regexp above "software description" field

File: services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/AbstractFeature.java [hunk 3]
```diff
@@ -20,10 +20,11 @@
 import net.floodlightcontroller.core.IOFSwitch;
 
 import java.util.Optional;
+import java.util.regex.Pattern;
 
 public abstract class AbstractFeature {
     protected static final String MANUFACTURER_NICIRA = "Nicira, Inc.";
-    protected static final String MANUFACTURER_NOVIFLOW = "NoviFlow Inc";
+    protected static final Pattern MANUFACTURER_NOVIFLOW = Pattern.compile("^NW\\d{3}\\.\\d+\\.\\d+$");
```
Comment (niksv) [inline services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/AbstractFeature.java]: I think we need unit tests for this regexp

Comment (surabujin) [reply services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/feature/AbstractFeature.java]: This is definetelly not "MANUFACTURER_NOVIFLOW". Keep this regexp on BfdFeature "level" and drop this constant if it is not used any more.

**Pull request merged**